rct229/rulesets/ashrae9012019/ruleset_functions/get_zone_BPF_BAT.py

```diff
@@ -0,0 +1,144 @@
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_MEDICAL_SUPPLY_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_NURSERY: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_NURSES_STATION: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_OPERATING_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_PATIENT_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_PHYSICAL_THERAPY_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.HEALTHCARE_FACILITY_RECOVERY_ROOM: "HEALTHCARE_HOSPITAL",
+    LIGHTING_SPACE.LIBRARY_READING_AREA: "UNDETERMINED",
+    LIGHTING_SPACE.LIBRARY_STACKS: "UNDETERMINED",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_DETAILED_MANUFACTURING_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_EQUIPMENTROOM: "ALL_OTHER",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_EXTRA_HIGH_BAY_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_HIGH_BAY_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MANUFACTURING_FACILITY_LOW_BAY_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MUSEUM_GENERAL_EXHIBITION_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.MUSEUM_RESTORATION_ROOM: "ALL_OTHER",
+    LIGHTING_SPACE.POST_OFFICE_SORTING_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.RELIGIOUS_FACILITY_FELLOWSHIP_HALL: "ALL_OTHER",
+    LIGHTING_SPACE.RELIGIOUS_FACILITY_WORSHIP_PULPIT_CHOIR_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.RETAIL_FACILITIES_DRESSING_FITTING_ROOM: "RETAIL",
+    LIGHTING_SPACE.RETAIL_FACILITIES_MALL_CONCOURSE: "RETAIL",
+    LIGHTING_SPACE.SPORTS_ARENA_PLAYING_AREA_CLASS_I_FACILITY: "ALL_OTHER",
+    LIGHTING_SPACE.SPORTS_ARENA_PLAYING_AREA_CLASS_II_FACILITY: "ALL_OTHER",
+    LIGHTING_SPACE.SPORTS_ARENA_PLAYING_AREA_CLASS_III_FACILITY: "ALL_OTHER",
+    LIGHTING_SPACE.SPORTS_ARENA_PLAYING_AREA_CLASS_IV_FACILITY: "UNDETERMINED",
+    LIGHTING_SPACE.TRANSPORTATION_FACILITY_BAGGAGE_CAROUSEL_AREA: "ALL_OTHER",
+    LIGHTING_SPACE.TRANSPORTATION_FACILITY_AIRPORT_CONCOURSE: "ALL_OTHER",
+    LIGHTING_SPACE.TRANSPORTATION_FACILITY_TICKET_COUNTER: "ALL_OTHER",
+    LIGHTING_SPACE.WAREHOUSE_STORAGE_AREA_MEDIUM_TO_BULKY_PALLETIZED_ITEMS: "WAREHOUSE",
+    LIGHTING_SPACE.WAREHOUSE_STORAGE_AREA_SMALLER_HAND_CARRIED_ITEMS: "WAREHOUSE",
+}
+
+
+def get_zone_BPF_BAT(rmd: dict, zone_id: str) -> dict[str, str]:
+    """
+     Get a dictionary of the BPF_BAT and areas for a given zone.
+
+     Parameters
+    ----------
+    RMD: str, The ruleset model descriptions
+    zone_id: str, The id of the zone
+
+    Returns
+    -------
+    zone_BPF_BAT_dict: A dict for the zone that saves the BPF_BAT as keys and the areas as the values.
+                       Example: {SCHOOL: 50000, UNDETERMINED: 2000}
+
+    """
+
+    zone_BPF_BAT_dict = {}
+    for space in find_all(
+        f'$.buildings[*].building_segments[*].zones[*][?(@.id = "{zone_id}")].spaces[*]',
+        rmd,
+    ):
+        lighting_space_type = getattr_(space, "spaces", "lighting_space_type")
```

Comment:
I don't think we want an undetermined outcome for the rule that calls this function if a space does not have a `lighting_space_type` defined. Instead, I think this should map the space area to the `"UNDETERMINED"` building area

Reply:
The RDS says ```space_BPF_BAT = building_area_lookup[space.lighting_space_type]```, which assumes the ```lighting_space_type``` key is always available. 

I can modify RDS when the ```lighting_space_type``` key isn't defined, the space type is mapped to ```UNDETERMINED```, but I just wanted to double-check with you.

Reply:
This issue was fixed!